source/MRMesh/MRVoxelsVolumeAccess.h

```diff
@@ -82,10 +96,63 @@ class VoxelsVolumeAccessor<VoxelsVolume<VoxelValueGetter<T>>>
         return data_( pos );
     }
 
+    ValueType safeGet( const Vector3i& pos ) const
+    {
+        return get( pos ); // assume bounds checking is handled by the getter
+    }
+
 private:
     const VoxelValueGetter<T>& data_;
 };
 
+/// accessor override with a preset background value: out-of bounds indexes correspond to a fixed value
+template <typename Accessor>
+class VoxelsVolumeAccessorWithBackground : public Accessor
+{
+public:
+    using VolumeType = typename Accessor::VolumeType;
+    using ValueType = typename VolumeType::ValueType;
+
+    VoxelsVolumeAccessorWithBackground( const VolumeType& volume, ValueType bgValue )
+        : Accessor( volume ), dims_( volume.dims ), bgValue_( bgValue )
+    {}
+
+    ValueType safeGet( const Vector3i& pos ) const
+    {
+        if ( pos.x < 0 || pos.y < 0 || pos.z < 0 ||
+             pos.x >= dims_.x || pos.y >= dims_.y || pos.z >= dims_.z )
+            return bgValue_;
+        return Accessor::get( pos );
+    }
+
+private:
+    Vector3i dims_;
+    ValueType bgValue_;
+};
+
+/// accessor override with repeating edge values for out-of-bound indexes
+template <typename Accessor>
+class VoxelsVolumeAccessorWithRepeat : public Accessor
```

Comment:
better use Clamp term here instead of Repeat, which has different meaning in OpenGL:
```
    case WrapType::Clamp:
        wrap = GL_CLAMP_TO_EDGE;
        break;
    case WrapType::Repeat:
        wrap = GL_REPEAT;
        break;
    case WrapType::Mirror:
        wrap = GL_MIRRORED_REPEAT;
        break;